modules/coach/src/main/CoachPager.scala

```diff
@@ -18,17 +23,64 @@ final class CoachPager(
   import CoachPager._
   import BsonHandlers._
 
-  def apply(lang: Option[Lang], order: Order, page: Int): Fu[Paginator[Coach.WithUser]] = {
-    val adapter = new Adapter[Coach](
-      collection = coll,
-      selector = listableSelector ++ lang.?? { l =>
-        $doc("languages" -> l.code)
-      },
-      projection = none,
-      sort = order.predicate
-    ) mapFutureList withUsers
+  def apply(
+      lang: Option[Lang],
+      order: Order,
+      country: Option[Country],
+      page: Int
+  ): Fu[Paginator[Coach.WithUser]] = {
+    def selector = listableSelector ++ lang.?? { l => $doc("languages" -> l.code) }
+
+    val adapter = country match {
+      case Some(country) => new AdapterLike[Coach.WithUser] {
+        def nbResults: Fu[Int] = coll.secondaryPreferred.countSel(selector)
+
+        def slice(offset: Int, length: Int): Fu[List[Coach.WithUser]] =
```

Comment:
it doesn't seem like `offset` is ever used? How does pagination work then? Can you load several pages of coaches?